### PR TITLE
minor functionality change: adding API functionality to tavilysearch

### PR DESCRIPTION
**Pull request by isahers1**

### CODE REVIEW

File: libs/community/langchain_community/tools/tavily_search/tool.py [hunk 1]
```diff
@@ -29,7 +29,14 @@ class TavilySearchResults(BaseTool):
     )
     api_wrapper: TavilySearchAPIWrapper = Field(default_factory=TavilySearchAPIWrapper)  # type: ignore[arg-type]
     max_results: int = 5
+    search_depth: Optional[str] = "advanced"
+    include_domains: Optional[List[str]] = []
+    exclude_domains: Optional[List[str]] = []
+    include_answer: Optional[bool] = False
+    include_raw_content: Optional[bool] = False
+    include_images: Optional[bool] = False
```
Comment (baskaryan) [inline libs/community/langchain_community/tools/tavily_search/tool.py:37]: dont need to be Optional

Comment (baskaryan) [reply libs/community/langchain_community/tools/tavily_search/tool.py:37]: also would be nice to add docstrings for all these (can be in a follow up pr)

Comment (isahers1) [reply libs/community/langchain_community/tools/tavily_search/tool.py]: Should be fixed in commit I just pushed

**Review by baskaryan — APPROVED** [review]
awesome 🚀

**Comment by eyurtsev** [comment]
@baskaryan  / @isahers1 this issue might be related: https://github.com/langchain-ai/langchain/issues/23632

File: libs/community/langchain_community/tools/tavily_search/tool.py [hunk 2]
```diff
@@ -29,6 +29,19 @@ class TavilySearchResults(BaseTool):
     )
     api_wrapper: TavilySearchAPIWrapper = Field(default_factory=TavilySearchAPIWrapper)  # type: ignore[arg-type]
     max_results: int = 5
+    """Max search results to return, default is 5"""
+    search_depth: str = "advanced"
+    '''The depth of the search. It can be "basic" or "advanced"'''
+    include_domains: List[str] = []
```
Comment (eyurtsev) [inline libs/community/langchain_community/tools/tavily_search/tool.py:35]: This is missing an Optional as a possible value. 

 `limit_to_domains` would be a clearer name probably. I assume that `[]` and a `None` have entirely opposite behaviors here, so the None value should be made explicit in the type signature

Comment (baskaryan) [reply libs/community/langchain_community/tools/tavily_search/tool.py:35]: what's this have to do with callbacks?

**Comment by dsolito** [comment]
Hello, I don't know if it's the right place to post it but I cannot get all the information requested from the params.
Seems that only max_result is kept.

```python
from langchain_community.tools.tavily_search import TavilySearchResults, TavilyAnswer


tool_with_raw = TavilySearchResults(include_raw_content=True, max_results=1)
tool_with_raw_and_answer = TavilySearchResults(include_raw_content=True, include_answer=True, max_results=1)
tool_without_raw = TavilySearchResults(include_raw_content=False, max_results=1)


r1=tool_with_raw.invoke({'query': 'how to cook a steak?'})
print(r1)

r2=tool_without_raw.invoke({'query': 'how to cook a steak?'})
print(r2)

r3=tool_with_raw_and_answer.invoke({'query': 'how to cook a steak?'})
print(r3)
``` 

```python
[
    {
        'url': 'https://www.onceuponachef.com/recipes/how-to-cook-steak-on-the-stovetop.html',
        'content': 'Pan-Seared Steaks\nPan-searing is the best way to cook a steak, and it’s also the easiest!\nIngredients\nInstructions\nPair 
with\nNutrition Information\nPowered by\nThis website is written and produced for informational purposes only. When I do this again I will do for 5 
minutes but will turn off the heat on my cast Iron frying pan 2 minutes before and add butter and rosemary and garlic to get the steak more to our 
liking.\n I got a ribeye steak, heated the pan to the top heat and did everything like you mentioned, but after three minutes the steak was burned, 
on the other side the same happened. After doing some more research, I find you have to bring the steak to room temperature before you cook it and 
yiu have to snip the fat around the edges to keep it from curling. 22 Quick and Easy Recipes in 30 Minutes (or less) + 5 Chef Secrets To Make You A 
Better Cook!\nFind a Recipe\nHow To Cook Steak On The Stovetop\nThis post may contain affiliate links.'
    }
]
>>> r2=tool_without_raw.invoke({'query': 'how to cook a steak?'})
>>> print(r2)
[
    {
        'url': 'https://www.onceuponachef.com/recipes/how-to-cook-steak-on-the-stovetop.html',
        'content': 'Pan-Seared Steaks\nPan-searing is the best way to cook a steak, and it’s also the easiest!\nIngredients\nInstructions\nPair 
with\nNutrition Information\nPowered by\nThis website is written and produced for informational purposes only. When I do this again I will do for 5 
minutes but will turn off the heat on my cast Iron frying pan 2 minutes before and add butter and rosemary and garlic to get the steak more to our 
liking.\n I got a ribeye steak, heated the pan to the top heat and did everything like you mentioned, but after three minutes the steak was burned, 
on the other side the same happened. After doing some more research, I find you have to bring the steak to room temperature before you cook it and 
yiu have to snip the fat around the edges to keep it from curling. 22 Quick and Easy Recipes in 30 Minutes (or less) + 5 Chef Secrets To Make You A 
Better Cook!\nFind a Recipe\nHow To Cook Steak On The Stovetop\nThis post may contain affiliate links.'
    }
]
>>> r3=tool_with_raw_and_answer.invoke({'query': 'how to cook a steak?'})
>>> print(r3)
[
    {
        'url': 'https://www.onceuponachef.com/recipes/how-to-cook-steak-on-the-stovetop.html',
        'content': 'Pan-Seared Steaks\nPan-searing is the best way to cook a steak, and it’s also the easiest!\nIngredients\nInstructions\nPair 
with\nNutrition Information\nPowered by\nThis website is written and produced for informational purposes only. When I do this again I will do for 5 
minutes but will turn off the heat on my cast Iron frying pan 2 minutes before and add butter and rosemary and garlic to get the steak more to our 
liking.\n I got a ribeye steak, heated the pan to the top heat and did everything like you mentioned, but after three minutes the steak was burned, 
on the other side the same happened. After doing some more research, I find you have to bring the steak to room temperature before you cook it and 
yiu have to snip the fat around the edges to keep it from curling. 22 Quick and Easy Recipes in 30 Minutes (or less) + 5 Chef Secrets To Make You A 
Better Cook!\nFind a Recipe\nHow To Cook Steak On The Stovetop\nThis post may contain affiliate links.'
    }
]
```

**Pull request merged**